tests/dummy/app/controllers/logging-example/application-log.js

```diff
@@ -0,0 +1,375 @@
+import Ember from 'ember';
+import ListFormController from 'ember-flexberry/controllers/i-i-s-caseberry-logging-objects-application-log-l';
+const { getOwner } = Ember;
+
+export default ListFormController.extend({
+  /**
+  Flexberry Logging Service
+
+   @property flexberryLoggingService
+   @type String[]
+   */
+  flexberryLoggingService: Ember.inject.service('flexberry-logging'),
+
+  _messageNumber: 0,
+
+  logLevel: 0,
+
+  /**
+    Available test application level settings
+
+    @property settings
+    @type String[]
+   */
+  settings: [],
+
+  /**
+   Default choise in settings
+
+   @property text
+   @type String[]
+   */
+  text: '',
+
+  /**
+   List buttons describers for Supported actions
+   */
+  customButtons:  [{
+      buttonName: 'Assert',
+      buttonAction: 'assertAction',
+      buttonClasses: 'ui orange button'
+    }, {
+      buttonName: 'Error',
+      buttonAction: 'errorAction',
+      buttonClasses: 'ui orange button'
+    }, {
+      buttonName: 'Throw',
+      buttonAction: 'throwAction',
+      buttonClasses: 'ui orange button'
+    }, {
+      buttonName: 'Warn',
+      buttonAction: 'warnAction',
+      buttonClasses: 'ui teal button'
+
+    }, {
+      buttonName: 'Log',
+      buttonAction: 'logAction',
+      buttonClasses: 'ui teal button'
+    }, {
+      buttonName: 'Info',
+      buttonAction: 'infoAction',
+      buttonClasses: 'ui teal button'
+    }, {
+      buttonName: 'Debug',
+      buttonAction: 'debugAction',
+      buttonClasses: 'ui yellow button'
+
+    }, {
+      buttonName: 'Deprecation',
+      buttonAction: 'deprecationAction',
+      buttonClasses: 'ui yellow button'
+    }],
+
+  _router: undefined,
+
+  /**
+   *    Selected jQuery object, containing HTML of confirm modal dialog.
+   *
+   *    @property confirmModalDialog
+   *    @type Object
+   *    @default null
+   */
+  confirmModalDialog: undefined,
+
+  /**
+   *    Content to be displayed in confirm modal dialog.
+   *    It will be displayed only if some confirm needs.
+   *
+   *    @property confirmModalDialogContent
+   *    @type String
+   *    @default 't('components.flexberry-file.confirm-dialog-content')'
+   */
+  confirmModalDialogContent: '',
+
+  /**
+   *    Category to be passed in action confirm modal dialog.
+   *
+   *    @property confirmModalDialogCategory
+   *    @type String
+   *    @default ''
+   */
+  confirmModalDialogCategory: '',
+
+  init() {
+    this._router = getOwner(this).lookup('router:main');
```

Comment:
We do not use such initialization variant in Ember.
We write 'this.set('_router', getOwner(this).lookup('router:main'))'. Here and below.

Also from route we can get current router like 'this.get('router')'. Is there something similar for controller?